File: airbyte-commons/src/main/java/io/airbyte/commons/version/AirbyteProtocolVersion.java
/*
 * Copyright (c) 2023 Airbyte, Inc., all rights reserved.
 */

package io.airbyte.commons.version;

public class AirbyteProtocolVersion {

  public static final Version DEFAULT_AIRBYTE_PROTOCOL_VERSION = new Version("0.2.0");
  public static final Version V0 = new Version("0.3.0");
  public static final Version V1 = new Version("1.0.0");

  public static final String AIRBYTE_PROTOCOL_VERSION_MAX_KEY_NAME = "airbyte_protocol_version_max";
  public static final String AIRBYTE_PROTOCOL_VERSION_MIN_KEY_NAME = "airbyte_protocol_version_min";

  public static Version getWithDefault(final String version) {
    if (version == null || version.isEmpty() || version.isBlank()) {
      return DEFAULT_AIRBYTE_PROTOCOL_VERSION;
    } else {
      return new Version(version);
    }
  }
}

File: airbyte-cdk/java/airbyte-cdk/dependencies/src/main/kotlin/io/airbyte/commons/version/AirbyteProtocolVersion.kt
/*
 * Copyright (c) 2023 Airbyte, Inc., all rights reserved.
 */
package io.airbyte.commons.version

object AirbyteProtocolVersion {
    val DEFAULT_AIRBYTE_PROTOCOL_VERSION: Version = Version("0.2.0")
    val V0: Version = Version("0.3.0")
    val V1: Version = Version("1.0.0")

    const val AIRBYTE_PROTOCOL_VERSION_MAX_KEY_NAME: String = "airbyte_protocol_version_max"
    const val AIRBYTE_PROTOCOL_VERSION_MIN_KEY_NAME: String = "airbyte_protocol_version_min"

    fun getWithDefault(version: String?): Version {
        return if (version == null || version.isEmpty() || version.isBlank()) {
            DEFAULT_AIRBYTE_PROTOCOL_VERSION
        } else {
            Version(version)
        }
    }
}
